authentication/views.py

```diff
@@ -49,13 +49,37 @@ def get_access_token(request):
                               public_key=PLAID_PUBLIC_KEY, environment=PLAID_ENV)
         public_token = request.POST.get('public_token')
         exchange_response = client.Item.public_token.exchange(public_token)
+        plaidrequest = client.Item.get(exchange_response['access_token'])
         bank_user = UserBank(
             user=request.user,
             item_id=exchange_response['item_id'],
-            access_token=exchange_response['access_token']
+            access_token=exchange_response['access_token'],
+            institution_name=plaidrequest['item']['institution_id'],
             )
         bank_user.save()
         request.user.profile.has_bank_linked = True
         request.user.save()
         return HttpResponseRedirect("/home")
     return HttpResponse("Please don't sniff urls")
+
+
+@login_required
+def list_transactions(request):
+    """
+    A demo view to list all of a user's transactions
+    """
+    if request.method == "POST":
+        user_bank_id = request.POST.get('id')
+        user_bank = UserBank.objects.get(pk=user_bank_id)
+        if user_bank.user == request.user:
+            client = plaid.Client(client_id=PLAID_CLIENT_ID, secret=PLAID_SECRET,
+                                  public_key=PLAID_PUBLIC_KEY, environment=PLAID_ENV)
+            response = client.Transactions.get(user_bank.access_token,
+                                               start_date='2016-07-12', end_date='2017-01-09')
```

Comment:
Hardcoded dates?

Reply:
yea, i gotta fix that. This was just pulling initial tx data; I'll be switching to yearly from present day next.